src/view.rs

```diff
@@ -859,3 +859,87 @@ mod view_response_builder_tests {
         assert_eq!(expected, got);
     }
 }
+
+pub struct AllDocumentsViewValue {
+    rev: Revision,
+}
+
+impl AllDocumentsViewValue {
+    pub fn revision(&self) -> &Revision {
+        &self.rev
+    }
+}
+
+impl serde::Serialize for AllDocumentsViewValue {
+    fn serialize<S>(&self, serializer: &mut S) -> Result<(), S::Error>
+        where S: serde::Serializer
+    {
+        Ok(try!(serializer.serialize_struct_elt("rev", &self.rev)))
+    }
+}
+
+impl serde::Deserialize for AllDocumentsViewValue {
```

Comment:
No doubt, writing them by hand avoids a whole lot of problems. I just find it hard to believe that you'd need that many lines for a seemingly simple task. I recently implemented a deserializer in Elm and that took a single line per field, with static typing and all (although it compiles to JS, maybe that's why). I'm probably just spoiled by dynamically typed languages. :)